src/functional_pipeline/functional_pipeline.m

```diff
@@ -62,20 +65,63 @@
 
 
 % PARSE INPUT
-[subjectDir, varargin{:}] = convertStringsToChars(subjectDir, varargin{:});
+[subjectSelection, varargin{:}] = convertStringsToChars(subjectSelection, varargin{:});
 
-assert(ischar(subjectDir), ...
+assert(ischar(subjectSelection), ...
     'CATO:functional_pipeline:subjectDirNotText', ...
     'subjectDir must be a row vector of characters or string scalar.');
-assert(isdir(subjectDir), ...
+assert(isdir(subjectSelection) | isfile(subjectSelection), ...
     'CATO:functional_pipeline:subjectDirNotDir', ...
-    'subjectDir (%s) is not a directory.', subjectDir);
+    'subjectDir (%s) is not a directory.', subjectSelection);
 
-[configFile, runType, configParamsCl] = parseVarargin(varargin{:});
-% parseVarargin does also error handling.
+%% Group mode
+
+if isfile(subjectSelection)
+    subjectListFile = subjectSelection;
+    fprintf('\n--------strarting group run----------\n')
+
+    % read file
+    fi = fopen(subjectListFile);
```

Comment:
Instead of using `fopen` and `fgetl` functions, we can maybe use the `readcell` function to read the subject list file. `readcell` handles trailing white spaces at the end of lines (which occurs often accidentally) and provides better error handling, which makes it a more reliable option.

Additionally, using `readcell` allows us to load all subject directories at once and let the pipeline check that they are valid before running the pipeline. This way, the user can get information about the number of valid subjects and catch errors before running the long pipeline.

Maybe, `readcell` has some drawbacks that I missed, in that case these issues could also be addressed with e.g. `deblank`.

Small note: The file handle is commonly referred to as `fid` instead of `fi` in CATO.